jobs.v3.api_client.email_alert_search_sample_test: test_email_alert_search_sample failed

Note: #5243 was also for this test, but it was closed more than 10 days ago. So, I didn't mark it flaky.

----

commit: 455f574254aa80d64b13ba84c3046824627e3c32
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/2dc61022-ec9c-4c27-a3e6-71cb3f37cf5b), [Sponge](http://sponge2/2dc61022-ec9c-4c27-a3e6-71cb3f37cf5b)
status: failed
<details><summary>Test output</summary><br><pre>Traceback (most recent call last):
  File "/workspace/jobs/v3/api_client/.nox/py-3-7/lib/python3.7/site-packages/backoff/_sync.py", line 105, in retry
    seconds = _next_wait(wait, jitter, elapsed, max_time_)
  File "/workspace/jobs/v3/api_client/.nox/py-3-7/lib/python3.7/site-packages/backoff/_common.py", line 27, in _next_wait
    value = next(wait)
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/workspace/jobs/v3/api_client/email_alert_search_sample_test.py", line 44, in test_email_alert_search_sample
    eventually_consistent_test()
  File "/workspace/jobs/v3/api_client/.nox/py-3-7/lib/python3.7/site-packages/backoff/_sync.py", line 108, in retry
    raise e
  File "/workspace/jobs/v3/api_client/.nox/py-3-7/lib/python3.7/site-packages/backoff/_sync.py", line 94, in retry
    ret = target(*args, **kwargs)
  File "/workspace/jobs/v3/api_client/email_alert_search_sample_test.py", line 42, in eventually_consistent_test
    assert re.search(expected, out)
AssertionError: assert None
 +  where None = <function search at 0x7f0beb94ae60>('.*matchingJobs.*', "{'metadata': {'requestId': '9786a60b-5ee3-44d1-a3f1-c5c60c91b372:APAb7ITx18H6HkqW3nDPOJsvBBjAVAmCtQ=='}}\n")
 +    where <function search at 0x7f0beb94ae60> = re.search</pre></details>

The log says:

```
------------------------------ Captured log call -------------------------------
INFO     backoff:_common.py:86 Backing off eventually_consistent_test(...) for 0.1s (AssertionError: assert None
 +  where None = <function search at 0x7f0beb94ae60>('.*matchingJobs.*', "{'metadata': {'requestId': '5fdba28c-baff-417c-b1f7-378491905303:APAb7IR1udwp9u6SVaHqdAlt0Ky/aF3YDQ=='}}\n")
 +    where <function search at 0x7f0beb94ae60> = re.search)
ERROR    backoff:_common.py:101 Giving up eventually_consistent_test(...) after 2 tries (StopIteration)
```

The backoff module gave up after 2 tries.

I think my usage of `retry_delay` function is wrong. I'll switch to the standard `backoff.expo`.

--- a/jobs/v3/api_client/email_alert_search_sample_test.py
+++ b/jobs/v3/api_client/email_alert_search_sample_test.py
@@ -26,15 +26,10 @@ def company_name():
     email_alert_search_sample.tear_down(company_name, job_name)
 
 
-def retry_delay():
-    # Always wait 10 seconds
-    yield 10
-
-
 @pytest.mark.flaky(max_runs=2, min_passes=1)
 def test_email_alert_search_sample(company_name, capsys):
 
-    @backoff.on_exception(retry_delay, AssertionError, max_time=600)
+    @backoff.on_exception(backoff.expo, AssertionError, max_time=120)
     def eventually_consistent_test():
         email_alert_search_sample.run_sample(company_name)
         out, _ = capsys.readouterr()

